cpu/esp_common/include/cpu.h

```diff
@@ -30,13 +30,14 @@ extern "C" {
 #define PROVIDES_PM_SET_LOWEST
 
 /**
- * @brief   Print the last instruction's address
+ * @brief   Gets the last instruction's address
  *
  * @todo:   Not supported
  */
-static inline void cpu_print_last_instruction(void)
+static inline uintptr_t cpu_get_last_instruction(void)
 {
     /* This function must exist else RIOT won't compile */
+    return 0;
```

Comment:
The following seems to be a bit simpler:
```suggestion
    register uintptr_t addr __asm__ ("a0");
    return addr;
```
However, `a0` gives the return address, that is the address of the instruction after the call of `_assert_panic`. To get the address of the call instruction of `assert_panic` itself, it would have to be:
```suggestion
    register uintptr_t addr __asm__ ("a0");
    return addr - 3;
```
Getting the address of the instruction before the call of `assert_panic` seems to be not possible since this instruction could have 16 or 24 bit length. Another problem is that ESP can uses the CALL0 ABI (ESP8266) and Windowed Register ABI (ESP32). In latter case `call8` instruction is used and the address in `a0` gives something `0x8.......` which corresponds to real address`0x4.......`. To solve it the right implementation would be:
```suggestion
    register uintptr_t addr __asm__ ("a0");
    return (addr >= 0x80000000) ? addr - 0x40000003 : addr - 3;
```

Reply:
I think the address of `__assert_panic` is good enough since it will still be the same line as the assert-macro that failed maybe the function name is a bit missleading (it returns the call most cases).
and for the adress we could just remove the upper 4 bits of a0 with `addr & 0x0fffffff`